samples/KitchenSink/Program.cs

```diff
@@ -2,7 +2,9 @@
 using System;
 using Extism;
 using System.Text.Json;
+using SampleLib;
 
+Class1.noop(); // Import Class1 from SampleLib so that it's included during compilation
```

Comment:
Seems like the library has to be called from the app, otherwise there will be a compilation error:
```
PS D:\dylibso\dotnet-pdk\samples\KitchenSink> dotnet build
MSBuild version 17.8.0+6cdef4241 for .NET
  Determining projects to restore...
  All projects are up-to-date for restore.
C:\Program Files\dotnet\sdk\8.0.100-rc.2.23502.2\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.RuntimeIdentifierInfe
rence.targets(311,5): message NETSDK1057: You are using a preview version of .NET. See: https://aka.ms/dotnet-suppo 
rt-policy [D:\dylibso\dotnet-pdk\samples\KitchenSink\KitchenSink.csproj]
C:\Program Files\dotnet\sdk\8.0.100-rc.2.23502.2\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.RuntimeIdentifierInfe
rence.targets(311,5): message NETSDK1057: You are using a preview version of .NET. See: https://aka.ms/dotnet-suppo 
rt-policy [D:\dylibso\dotnet-pdk\samples\SampleLib\SampleLib.csproj]
  Extism.Pdk -> D:\dylibso\dotnet-pdk\src\Extism.Pdk\bin\Debug\net8.0\wasi-wasm\Extism.Pdk.dll
  SampleLib -> D:\dylibso\dotnet-pdk\samples\SampleLib\bin\Debug\net8.0\SampleLib.dll
D:\dylibso\dotnet-pdk\samples\KitchenSink\Program.cs(94,25): warning CS8618: Non-nullable property 'Parts' must con
tain a non-null value when exiting constructor. Consider declaring the property as nullable. [D:\dylibso\dotnet-pdk 
\samples\KitchenSink\KitchenSink.csproj]
D:\dylibso\dotnet-pdk\samples\KitchenSink\Program.cs(27,27): warning IL2026: Using member 'System.Text.Json.JsonSer 
ializer.Deserialize<TValue>(ReadOnlySpan<Byte>, JsonSerializerOptions)' which has 'RequiresUnreferencedCodeAttribut 
e' can break functionality when trimming application code. JSON serialization and deserialization might require typ 
es that cannot be statically analyzed. Use the overload that takes a JsonTypeInfo or JsonSerializerContext, or make 
 sure all of the required types are preserved. [D:\dylibso\dotnet-pdk\samples\KitchenSink\KitchenSink.csproj]       
  KitchenSink -> D:\dylibso\dotnet-pdk\samples\KitchenSink\bin\Debug\net8.0\wasi-wasm\KitchenSink.dll
  MicrosoftNetCoreAppRuntimePackRidNativeDir: C:\Program Files\dotnet\packs\Microsoft.NETCore.App.Runtime.Mono.wasi 
  -wasm\8.0.0-rc.2.23479.6\runtimes\wasi-wasm\native\
  Performing WASI SDK build: "D:\tools\wasi_sdk\bin\clang.exe" "@D:\dylibso\dotnet-pdk\samples\KitchenSink\obj\Debu
  g\net8.0\wasi-wasm\wasm\for-build\clang-compile.rsp"
wasm-ld : error : C:/Users/muham/AppData/Local/Temp/pinvoke-e1a597.o: undefined symbol: samplelib_import [D:\dylibs
o\dotnet-pdk\samples\KitchenSink\KitchenSink.csproj]
clang : error : linker command failed with exit code 1 (use -v to see invocation) [D:\dylibso\dotnet-pdk\samples\Ki
tchenSink\KitchenSink.csproj]
C:\Program Files\dotnet\packs\Microsoft.NET.Runtime.WebAssembly.Wasi.Sdk\8.0.0-rc.2.23479.6\Sdk\WasiApp.Native.targ
ets(478,5): error MSB3073: The command ""D:\tools\wasi_sdk\bin\clang.exe" "@D:\dylibso\dotnet-pdk\samples\KitchenSi 
nk\obj\Debug\net8.0\wasi-wasm\wasm\for-build\clang-compile.rsp"" exited with code 1. [D:\dylibso\dotnet-pdk\samples 
\KitchenSink\KitchenSink.csproj]

Build FAILED.

D:\dylibso\dotnet-pdk\samples\KitchenSink\Program.cs(94,25): warning CS8618: Non-nullable property 'Parts' must con 
tain a non-null value when exiting constructor. Consider declaring the property as nullable. [D:\dylibso\dotnet-pdk 
\samples\KitchenSink\KitchenSink.csproj]
D:\dylibso\dotnet-pdk\samples\KitchenSink\Program.cs(27,27): warning IL2026: Using member 'System.Text.Json.JsonSer 
ializer.Deserialize<TValue>(ReadOnlySpan<Byte>, JsonSerializerOptions)' which has 'RequiresUnreferencedCodeAttribut 
e' can break functionality when trimming application code. JSON serialization and deserialization might require typ 
es that cannot be statically analyzed. Use the overload that takes a JsonTypeInfo or JsonSerializerContext, or make 
 sure all of the required types are preserved. [D:\dylibso\dotnet-pdk\samples\KitchenSink\KitchenSink.csproj]       
wasm-ld : error : C:/Users/muham/AppData/Local/Temp/pinvoke-e1a597.o: undefined symbol: samplelib_import [D:\dylibs 
o\dotnet-pdk\samples\KitchenSink\KitchenSink.csproj]
clang : error : linker command failed with exit code 1 (use -v to see invocation) [D:\dylibso\dotnet-pdk\samples\Ki 
tchenSink\KitchenSink.csproj]
C:\Program Files\dotnet\packs\Microsoft.NET.Runtime.WebAssembly.Wasi.Sdk\8.0.0-rc.2.23479.6\Sdk\WasiApp.Native.targ
ets(478,5): error MSB3073: The command ""D:\tools\wasi_sdk\bin\clang.exe" "@D:\dylibso\dotnet-pdk\samples\KitchenSi 
nk\obj\Debug\net8.0\wasi-wasm\wasm\for-build\clang-compile.rsp"" exited with code 1. [D:\dylibso\dotnet-pdk\samples 
\KitchenSink\KitchenSink.csproj]
    2 Warning(s)
    3 Error(s)

Time Elapsed 00:00:18.89
```